tools/runners/run-test-suite-test262.sh

```diff
@@ -14,108 +14,58 @@
-      [?])
-        usage ;;
-    esac
-done
+declare -a CHAPTERS_TO_TEST=("ch06" "ch07" "ch08" "ch09" "ch10" "ch11" "ch12" "ch13" "ch14" "ch15")
+declare -a SUBCHAPTERS_TO_TEST=("${CHAPTER07_TO_TEST[@]}" "${CHAPTER08_TO_TEST[@]}" "${CHAPTER09_TO_TEST[@]}" "${CHAPTER10_TO_TEST[@]}" "${CHAPTER11_TO_TEST[@]}" "${CHAPTER12_TO_TEST[@]}" "${CHAPTER13_TO_TEST[@]}" "${CHAPTER14_TO_TEST[@]}")
 
-TMP_DIR=`mktemp -d --tmpdir=./`
+declare -a CHAPTERS_WITH_SUBCHAPTERS=("${CHAPTERS_TO_TEST[@]}" "${SUBCHAPTERS_TO_TEST[@]}")
 
-echo $TMP_DIR | grep -q tmp || exit 1
-
-trap clean_on_exit INT
-
-function clean_on_exit() {
-  rm -rf $TMP_DIR
-
-  [[ $1 == "OK" ]] || exit 1
+function run_test262 () {
+  ./tools/packaging/test262.py --command "timeout 3s $1" --full-summary "$2" > "${REPORT_PREFIX}"."$2"
 }
 
-TEST_SUITE_PATH="./tests/test262/test"
-STA_JS="$TEST_SUITE_PATH/harness/sta-jerry.js"
-MAX_THREADS=${MAX_CPUS:-32}
-
-[ $VERBOSE_REPORT -eq 0 ] || rm -f jerry.error.*
-rm -f test262.report
-
-find $TEST_SUITE_PATH/suite -name *.js -printf "%p %D%i\0" | xargs -0 -n 1 -P $MAX_THREADS bash -c '
-  target=$0
-  sta_js_path=$1;
-  timeout=$2;
-  tmp_dir=$3;
-  test=`echo $4 | cut -d " " -f 1`;
-  test_id=`echo $4 | cut -d " " -f 2`;
-  chapter=`echo $test | cut -d "/" -f 6`;
-
-  grep "\* @negative" $test 2>&1 >/dev/null;
-  negative=$?;
-
-  cmd="$target $test $sta_js_path";
-  output=`ulimit -t $timeout; $cmd 2>&1`;
-  status=$?;
-  output=`echo "$output" | sed -e "s/^/ /"`
-
-  if [[ $status -eq 0 && $negative -eq 0 || $status -ne 0 && $negative -ne 0 ]];
-  then
-    (echo "====================";
-     echo "Run \"$cmd\" failed: $status";
-     echo "---------------------";
-     echo "$output";
-     echo;) >> $tmp_dir/jerry.error."$chapter"."$test_id";
-  else
-    (echo "====================";
-     echo "Run \"$cmd\" successful";
-     echo "---------------------";
-     echo "$output";
-     echo;) >> $tmp_dir/jerry.ok."$chapter"."$test_id";
-  fi;
-' "$TARGET $PARSE" $STA_JS $TIMEOUT $TMP_DIR 2>/dev/null;
-
-if [ $VERBOSE_REPORT -eq 1 ]
-then
-  for CHAPTER in `ls $TEST_SUITE_PATH/suite`;
-  do
-    cat $TMP_DIR/jerry.error."$CHAPTER".* > jerry.error."$CHAPTER"
-  done
-fi
+function show_report_results () {
+  echo ""
+  echo "Chapter $1:"
+  grep -A3 "=== Summary ===" "${REPORT_PREFIX}"."$1"
+  echo "==============="
+}
 
-cat $TMP_DIR/jerry.ok.* $TMP_DIR/jerry.error.* | grep "^Run" > test262.report
+echo "Starting test262 testing for ${ENGINE}."
 
-ok_num=`grep "successful$" test262.report | wc -l`
-error_num=`grep "Run .* failed: [0-9]*$" test262.report | wc -l`
+for TEST_NAME in "${CHAPTERS_WITH_SUBCHAPTERS[@]}"
+do
+  run_test262 "${ENGINE}" "$TEST_NAME" &
+  RUN_PIDS="$RUN_PIDS $!";
+done
 
-ok_percent=`echo $ok_num $error_num | awk '{print 100.0 * $1 / ($1 + $2);}'`
-error_percent=`echo $ok_num $error_num | awk '{print 100.0 * $2 / ($1 + $2);}'`
+for RUN_PIDS in $RUN_PIDS
+do
+  wait "$RUN_PIDS" || RESULT_OK=0
```

Comment:
I'm passing timeout via command `--command "timeout 3s $ARG_ENGINE"`.
Btw, it's good idea to make it variable or flag, thanks.